backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/codec/impl/ElasticsearchYearFieldCodec.java

```diff
@@ -19,4 +20,9 @@ public ElasticsearchYearFieldCodec(DateTimeFormatter delegate) {
 	protected Year nullUnsafeParse(String stringValue) {
 		return Year.parse( stringValue, formatter );
 	}
+
+	@Override
+	protected Long nullUnsafeScalar(Year value) {
+		return value.atDay( 1 ).atStartOfDay().atZone( ZoneOffset.UTC ).toInstant().toEpochMilli();
```

Comment:
You can just call `value.toInstant( ZoneOffset.UTC )` instead of `atZone(...).toInstant()`.